### PR TITLE
python/hamming: Include a recursive example

### PR DESCRIPTION
**Pull request by jeffdparker**
Discourage the use of tail recursion here.

### CODE REVIEW

**Comment by NobbZ** [comment]
That version is not tail recursive, but yes python does not support recursive functions very well and a loop should always be preferred over recursion.

**Comment by NobbZ** [comment]
I'm also not sure if this should be listed as "reasonable" solution with the remark to discourage it.

**Comment by jeffdparker** [comment]
It's not Tail Recursive?  Really?

**Comment by NobbZ** [comment]
Yes, it is not tail recursive, as tail recursiveness means that the last function call is to the function itself, but in your case, the last function called is `+`.

In FP it is common to create a helper function which carries an additional accumulating value or to use some kind of `reduce`/`fold`.

**Comment by NobbZ** [comment]
You are right on the notes not mentioning tail recursiveness, still I think it's incorrect to have the example as a reasonable solution with a comment discouraging it at the same time.

**Comment by cmccandless** [comment]
> still I think it's incorrect to have the example as a reasonable solution with a comment discouraging it at the same time.

Maybe it would be best to reword that comment, along the lines of

`This solution works, but will take more time and more stack space than an iterative solution. Encourage the student to think in terms of performance as well as correctness.`

File: tracks/python/exercises/hamming/mentoring.md [hunk 1]
```diff
@@ -63,10 +63,31 @@ def distance(strand_a, strand_b):
...
+Encourage the student to think in terms of performance as well as correctness.
+
 ### Common Suggestions
 
 A List Comprehension can be suggested, but isn't an obvious win.
 	 
 ### Talking points
 
 Running time is not a concern here. It is hard to make this worse than O(N).
+
```
Comment (cmccandless) [inline tracks/python/exercises/hamming/mentoring.md]: This extra trailing whitespace is unnecessary.

Comment (cmccandless) [reply tracks/python/exercises/hamming/mentoring.md]: I'm referring to the added 2 newlines at the end of the file.

**Pull request merged**